ds-caselaw-ingester/lambda_function.py

```diff
@@ -389,44 +384,45 @@ def handler(event, context):
 
     # Extract and parse the judgment XML
     xml_file_name = metadata["parameters"]["TRE"]["payload"]["xml"]
-    uri = extract_uri(metadata, consignment_reference)
-    print(f"Ingesting document {uri}")
-    xml = get_best_xml(uri, tar, xml_file_name, consignment_reference)
-
-    updated = update_judgment_xml(uri, xml)
-    inserted = False if updated else insert_document_xml(uri, xml)
-
-    if updated:
-        # Notify editors that a document has been updated
-        send_updated_judgment_notification(uri, metadata)
-        unpublish_updated_judgment(uri)
-        print(f"Updated judgment xml for {uri}")
-    elif inserted:
+    source_uri = extract_uri(metadata, consignment_reference)
+    target_uri = select_uri(source_uri)
+    print(f"Ingesting document {source_uri} at {target_uri}")
+    xml = get_best_xml(target_uri, tar, xml_file_name, consignment_reference)
+
+    inserted = insert_document_xml(target_uri, xml)
+
+    if inserted:
         # Notify editors that a new document is ready
-        send_new_judgment_notification(uri, metadata)
-        print(f"Inserted judgment xml for {uri}")
+        send_new_judgment_notification(target_uri, metadata)
+        print(f"Inserted judgment xml for {source_uri} at {target_uri}")
     else:
         raise DocumentInsertionError(
-            f"Judgment {uri} failed to insert into Marklogic. Consignment Ref: {consignment_reference}"
+            f"Judgment {source_uri} failed to insert into Marklogic at {target_uri}."
+            f"Consignment Ref: {consignment_reference}"
         )
 
     # Store metadata
-    store_metadata(uri, metadata)
+    store_metadata(target_uri, metadata)
 
     # Store docx and rename
     docx_filename = extract_docx_filename(metadata, consignment_reference)
     copy_file(
         tar,
         f"{consignment_reference}/{docx_filename}",
-        f'{uri.replace("/", "_")}.docx',
-        uri,
+        f'{target_uri.replace("/", "_")}.docx',
+        target_uri,
         s3_client,
     )
+    # TODO: the editor move functionality should change the name of the docx file to match?
```

Comment:
oh that's a good question. Probably makes sense to and also to do so when we move the document in in the merge flow in editor ui?

Reply:
This should probably block the merging of this.

Reply:
We should probably note that this is yet another good reason to distinguish between:

- Our own internal identifier for a document, which is really just a UUID, and
- Our 'TNA' identifier for a document, which is an arbitrary string we use in URIs, and
- Other identifiers people may use such as NCNs or Bailii numbers

Reply:
We do actually do this; I had to double check.